<launch>

  <node pkg="pepper_local_republisher" type="tf_transform_republisher_node" name="tf_transform_republisher_node" output="screen">
    <param name="publish_frequency"                   value="10" />
    <param name="static_tf_transform"                 value="true" />
    <param name="input_tf_transform_parent_frame_id"  value="/CameraTop_optical_frame" />
    <param name="input_tf_transform_child_frame_id"   value="/CameraDepth_optical_frame" />
    <param name="output_tf_transform_parent_frame_id" value="/CameraTop_optical_frame_republished" />
    <param name="output_tf_transform_child_frame_id"  value="/CameraDepth_optical_frame_republished" />
  </node>

  <node pkg="pepper_local_republisher" type="camera_info_republisher_node" name="pepper_front_camera_info_republisher_node" output="screen">
    <param name="publish_frequency"                 value="10" />
    <param name="static_camera_info"                value="true" />
    <param name="input_camera_info_topic_name"      value="/pepper_robot/camera/front/camera_info" />
    <param name="output_camera_info_topic_name"     value="/pepper_local_republisher/pepper_robot/camera/front/camera_info" />
    <param name="output_camera_info_frame_id_name"  value="/CameraTop_optical_frame_republished" />
  </node>

  <node pkg="pepper_local_republisher" type="camera_info_republisher_node" name="pepper_depth_camera_info_republisher_node" output="screen">
    <param name="publish_frequency"                 value="10" />
    <param name="static_camera_info"                value="true" />
    <param name="input_camera_info_topic_name"      value="/pepper_robot/camera/depth/camera_info" />
    <param name="output_camera_info_topic_name"     value="/pepper_local_republisher/pepper_robot/camera/depth/camera_info" />
    <param name="output_camera_info_frame_id_name"  value="/CameraDepth_optical_frame_republished" />
  </node>

  <node pkg="pepper_local_republisher" type="image_republisher_node" name="pepper_front_camera_republisher_node" output="screen">
    <param name="input_image_topic_name"    	value="/pepper_robot/camera/front/image_rect_color" />
    <param name="input_image_transport_type" 	value="compressed" />
    <param name="output_image_topic_name"   	value="/pepper_local_republisher/pepper_robot/camera/front/image_rect_color" />
    <param name="output_image_frame_id_name"	value="/CameraTop_optical_frame_republished" />
  </node>

  <node pkg="pepper_local_republisher" type="image_republisher_node" name="pepper_depth_camera_republisher_node" output="screen">
    <param name="input_image_topic_name"      value="/pepper_robot/camera/depth/image_rect" />
    <param name="input_image_transport_type" 	value="compressedDepth" />
    <param name="output_image_topic_name"     value="/pepper_local_republisher/pepper_robot/camera/depth/image_rect" />
    <param name="output_image_frame_id_name"	value="/CameraDepth_optical_frame_republished" />
  </node>

  <node pkg="pepper_local_republisher" type="laser_scan_republisher_node" name="pepper_laser_scan_republisher_node" output="screen">
    <param name="input_laser_scan_topic_name"   value="/pepper_robot/laser" />
    <param name="output_laser_scan_topic_name"  value="/pepper_local_republisher/pepper_robot/laser" />
  </node>

  <!-- Register depth image -->
  <node pkg="nodelet" type="nodelet" name="register_depth_image" args="standalone depth_image_proc/register">

    <remap from="rgb/camera_info" to="/pepper_local_republisher/pepper_robot/camera/front/camera_info"/>
    <remap from="depth/camera_info" to="/pepper_local_republisher/pepper_robot/camera/depth/camera_info"/>
    <remap from="depth/image_rect" to="/pepper_local_republisher/pepper_robot/camera/depth/image_rect"/>

    <remap from="depth_registered/camera_info" to="/pepper_local_republisher/pepper_robot/camera/depth_registered/sw_registered/camera_info"/>
    <remap from="depth_registered/image_rect" to="/pepper_local_republisher/pepper_robot/camera/depth_registered/sw_registered/image_rect"/>

    <param name="queue_size" value="10"/>

  </node>

  <!-- Convert to point cloud -->
  <node pkg="nodelet" type="nodelet" name="pointcloud_publisher" args="standalone depth_image_proc/point_cloud_xyzrgb">

    <param name="queue_size" value="10"/>

    <!-- Input: Rectified depth image, registered to the RGB camera. -->
    <remap from="depth_registered/image_rect" to="/pepper_local_republisher/pepper_robot/camera/depth_registered/sw_registered/image_rect"/> 
    <!-- Input: Rectified color image. -->
    <remap from="rgb/image_rect_color" to="/pepper_local_republisher/pepper_robot/camera/front/image_rect_color"/> 
    <!-- Input: Camera calibration and metadata. --> 
    <remap from="rgb/camera_info" to="/pepper_local_republisher/pepper_robot/camera/front/camera_info"/>
    <!-- Output: XYZRGB point cloud. -->
    <remap from="depth_registered/points" to="/pepper_local_republisher/pepper_robot/camera/depth_registered/points"/>
  </node>

</launch>
